src/main/java/parser/ArithmetikParserClass.java

```diff
@@ -0,0 +1,879 @@
+		this.parseTree = parseTree;
+		this.input = input;
+		this.pointer = 0;
+		this.maxPointer = input.length - 1;
+	}
+	
+	//-------------------------------------------------------------------------
+	//-------------------Methoden der Grammatik--------------------------------
+	//-------------------------------------------------------------------------
+	
+	//-------------------------------------------------------------------------
+	// program -> emojiStartCode sequence emojiEndCode
+	// Der Parameter sT ist die Wurzel des bis hier geparsten Syntaxbaumes
+	//-------------------------------------------------------------------------
+	boolean program(SyntaxTree sT){
+		byte [] emojiStartCodeSet = {EMOJI_START_CODE};
+		byte [] emojiEndCodeSet = {EMOJI_END_CODE};
+		byte [] identSet = {IDENT};
+
+		// Falls aktuelle Eingabe ':triangular_flag_on_post:'
+		if (match(emojiStartCodeSet,sT)) {
+    		if (sequence(sT.insertSubtree(SEQUENCE))){
+    			if(match(emojiEndCodeSet,sT)) {
+    				return true;
+    			}else{//Syntaxfehler
+					syntaxError("EMOJI_END_OF_CODE erwartet"); 			
+ 					return false;
+    			}
+    		}else{//Syntaxfehler
+    			syntaxError("Fehler in geschachtelter Sequence"); 			
+ 				return false;
+    		}
+		}{//Syntaxfehler
+			syntaxError("EMOJI_start_OF_CODE erwartet"); 			
+				return false;
+		}
+	}//program
+	
+	//-------------------------------------------------------------------------
+	// sequence -> '{' instruction '}'
+	// Der Parameter sT ist die Wurzel des bis hier geparsten Syntaxbaumes
+	//-------------------------------------------------------------------------
+	boolean sequence(SyntaxTree sT){
+		byte [] openCurlySet= {OPEN_CURLY_BRACKET};
+		byte [] closeCurlySet= {CLOSE_CURLY_BRACKET};
+		byte [] numSet={NUM};
+		byte [] identSet={IDENT};
+
+		if (match(openCurlySet,sT)) {
+    		if (instruction(sT.insertSubtree(INSTRUCTION))){
+    			if(match(closeCurlySet,sT)) {
+    				return true;
+    			}else{//Syntaxfehler
+					syntaxError("Geschlossene geschweifte Klammer erwartet"); 			
+ 					return false;
+    			}
+    		}else{
+    			syntaxError("Fehler in geschachtelter Expression"); 			
+ 				return false;
+    		}
+		}
+		return false;
+	}//sequence
+	
+	
+	//-------------------------------------------------------------------------
+	// instruction -> 	variableDefinition instruction |
+	//					assigment instruction | 
+	//					while instruction | 
+	//					if instruction | 
+	//					for instruction | 
+	//					epsilon?
+	// Der Parameter sT ist die Wurzel des bis hier geparsten Syntaxbaumes
+	//-------------------------------------------------------------------------
+	boolean instruction(SyntaxTree sT){
+		byte [] identSet = {IDENT};
+		byte [] emojiWhileSet = {EMOJI_WHILE};
+		byte [] emojiIfSet = {EMOJI_IF};
+		byte [] emojiForSet = {EMOJI_FOR};
+		
+		if(matchDoesNotMovePointer(identSet,sT)) {
+			return assigment(sT.insertSubtree(ASSIGMENT)) && instruction(sT.insertSubtree(INSTRUCTION));
```

Comment:
Nur als reminder: Alles was ich vorhin angemerkt habte, bezieht sich auf den gesamten Code.

So was sollte auf jeden Fall in einer separaten Methode geprüft werden, die einen aussagekräftigen Namen. Mir ist jetzt auf Anhieb nicht klar worauf geprüft wird.